joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeRecovery.scala

```diff
@@ -1185,7 +1187,11 @@ abstract class RecoverForXCompilationUnit[CompilationUnitType <: AstNode](
   private def storeNodeTypeInfo(storedNode: StoredNode, types: Seq[String]): Unit = {
     lazy val existingTypes = storedNode.getKnownTypes
 
-    if (types.nonEmpty && types.toSet != existingTypes) {
+    val hasUnknownTypeFullName = storedNode
+      .property(PropertyNames.TYPE_FULL_NAME, "ANY")
```

Comment:
x2cpg.Defines has a constant for this now
```suggestion
      .property(PropertyNames.TYPE_FULL_NAME, Defines.Any)
```